agent/consul/gateways/gateway_meta_test.go

```diff
@@ -0,0 +1,409 @@
+package gateways
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetGatewayMeta(t *testing.T) {
```

Comment:
are these tests safe to run in parallel? if so can we add a `t.ParallelI()` here (and in the `t.Run`, we just need to make sure to capture the `tc` loop variable) https://pkg.go.dev/testing#T.Parallel